src/traces/carpet/calc.js

```diff
@@ -52,8 +52,8 @@ module.exports = function calc(gd, trace) {
     trace.setScale();
 
     // Convert cartesian-space x/y coordinates to screen space pixel coordinates:
-    t.xp = trace.xp = map2dArray(trace.xp, x, xa.c2p);
-    t.yp = trace.yp = map2dArray(trace.yp, y, ya.c2p);
+    t.xp = trace._xp = map2dArray(trace._xp, x, xa.c2p);
+    t.yp = trace._yp = map2dArray(trace._yp, y, ya.c2p);
```

Comment:
Are these `xp` and `yp` values used anywhere? [Here](https://github.com/plotly/plotly.js/blob/3d65de3adb20ca0c75e9ecb67739c5d801cc3b24/src/traces/carpet/plot.js#L81-L82) they appear to be reset during the plot step.